splunk-otel-android/src/main/java/com/splunk/rum/ErrorIdentifierExtractor.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import android.app.Application;
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.util.Log;
+import androidx.annotation.Nullable;
+
+public class ErrorIdentifierExtractor {
+
+    private static final String SPLUNK_UUID_MANIFEST_KEY = "SPLUNK_OLLY_CUSTOM_UUID";
+    @Nullable private static ErrorIdentifierExtractor instance = null;
+
+    private final Context context;
+    private final PackageManager packageManager;
+    private final ApplicationInfo applicationInfo;
```

Comment:
I think these 3 fields don't really need to be on this class...and it's a symptom of the class being more than one thing.